### PR TITLE
Update woocommerce sign in and sign up screens 

### PR DESCRIPTION
**Pull request by chihsuan**
## Proposed Changes

* Implement new UI design for Woo sign in and sign up screens (figma: 4ixWMlzrxllx93tSFsCW6k-fi-2656%3A36408)
  - Move all `.woo` selector  CSS styles from `client/layout/masterbar/oauth-client.scss` to `client/layout/masterbar/woo.scss` and update it for new design.
  - Add/update terms for Woo

### Signup

Before:

Screen Shot 2022-09-23 at 15 40 53

After:

Screen Shot 2022-09-23 at 15 41 17
### Login

Before:

Screen Shot 2022-09-23 at 15 42 03

After:

Screen Shot 2022-09-23 at 17 09 27

### 2FA

Before:
Screen Shot 2022-09-23 at 15 44 21

Screen Shot 2022-09-23 at 15 44 34

After:
Screen Shot 2022-09-23 at 15 45 48

Screen Shot 2022-09-23 at 15 45 55

Note that if the email address is already used, we should redirect user to the login form. I'll implement it in another PR.

## Testing Instructions

Set up a local `wpcalypso.wordpress.com` environment https://github.com/Automattic/wp-calypso/pull/67537#pullrequestreview-1101745311

**Signup**

1. Log out WordPress.com or use incognito mode
2. Go to https://woocommerce.com/start/?nuxentrysource=signup_menu
3. Replace the URL https://wordpress.com with https://wpcalypso.wordpress.com
4. Observe that the screen matches the design 4ixWMlzrxllx93tSFsCW6k-fi-2656%3A36408
5. Please also test in different screen sizes/browsers.

**Login & 2FA**

1. Log out WordPress.com or use incognito mode
2. Go to https://woocommerce.com/my-account/
3.  Replace the URL https://wordpress.com with https://wpcalypso.wordpress.com
4. Observe that the screen matches the design 4ixWMlzrxllx93tSFsCW6k-fi-2656%3A40038
5. Copy the login URL
6. Log in WordPress.com
7. Enable your 2FA security key
8. Go to the login screen via the Login URL 
9. Observe that the screen matches the design 4ixWMlzrxllx93tSFsCW6k-fi-2656%3A36505
10. Fill out email and password
11. Click on "Get started" button
12. Observe that 2FA screen matches the design 4ixWMlzrxllx93tSFsCW6k-fi-2656%3A39915
13. Click on "Continue with your authenticator app" button
14. Observe that the screen matches the design 4ixWMlzrxllx93tSFsCW6k-fi-2656%3A39973
15. Please also test in different screen sizes/browsers.

Please also confirm that these changes don't affect other screens:

WP signup
- https://wordpress.com/start/user?ref=logged-out-homepage-lp
- https://wpcalypso.wordpress.com/start/user?ref=logged-out-homepage-lp

WP login
- https://wordpress.com/log-in/new?redirect_to=https%3A%2F%2Fwordpress.com%2Fstart%2Fdomains%3Fref%3Dlogged-out-homepage-lp&signup_url=%2Fstart%2Fuser%3Fref%3Dlogged-out-homepage-lp
- https://wpcalypso.wordpress.com/log-in/new?redirect_to=https%3A%2F%2Fwpcalypso.wordpress.com%2Fstart%2Fdomains%3Fref%3Dlogged-out-homepage-lp&signup_url=%2Fstart%2Fuser%3Fref%3Dlogged-out-homepage-lp

WC partner signup

- https://woocommerce.com/partner-signup

WC core Jetpack connect the flow

- [wordpress.com/jetpack/connect](https://wordpress.com/jetpack/connect/authorize?client_id=210832473&redirect_uri=https%3A%2F%2Ffinancial-spideri%2Fwp-admin%2Fadmin.php%3Fhandler%3Djetpack-connection-webhooks%26action%3Dauthorize%26_wpnonce%3D3ebbe29855%26redirect%3Dhttps%253A%252F%252Ffinancial-spider.jurassic.ninja%252Fwp-admin%252Fadmin.php%253Fpage%253Dwc-admin&state=1&scope=administrator%3Ab1ba3f3a29279c33aa54b534b37fb67a&user_email=chi-hsuan.huang%40automattic.com&user_login=chihsuanhuang&jp_version=11.3.2&secret=qsSO5buAhDbD9JSaBH1BhX3ML62lZEHY&blogname=Financial+Spider&site_url=https%3A%2F%2Ffinancial-spider.jurassic.ninja&home_url=https%3A%2F%2Ffinancial-spider.jurassic.ninja&site_icon&site_lang=en_US&site_created=2022-09-23+08%3A07%3A48&allow_site_connection=1&locale=en&_ui=woo%3AlxDZ4Sce8nm0xpiTaNFN3I6p&_ut=anon&calypso_env=production&_as=wp-cli&from=woocommerce-onboarding&purchase_nonce=qV47s0a6&_wp_nonce=7fe5053a58&redirect_after_auth=https%3A%2F%2Ffinancial-spider.jurassic.ninja%2Fwp-admin%2Fadmin.php%3Fpage%3Dwc-admin&site=https%3A%2F%2Ffinancial-spider.jurassic.ninja)
- [wpcalypso.wordpress.com/jetpack/connect](https://wpcalypso.wordpress.com/jetpack/connect/authorize?client_id=210832473&redirect_uri=https%3A%2F%2Ffinancial-spider%2Fwp-admin%2Fadmin.php%3Fhandler%3Djetpack-connection-webhooks%26action%3Dauthorize%26_wpnonce%3D3ebbe29855%26redirect%3Dhttps%253A%252F%252Ffinancial-spider.jurassic.ninja%252Fwp-admin%252Fadmin.php%253Fpage%253Dwc-admin&state=1&scope=administrator%3Ab1ba3f3a29279c33aa54b534b37fb67a&user_email=chi-hsuan.huang%40automattic.com&user_login=chihsuanhuang&jp_version=11.3.2&secret=qsSO5buAhDbD9JSaBH1BhX3ML62lZEHY&blogname=Financial+Spider&site_url=https%3A%2F%2Ffinancial-spider.jurassic.ninja&home_url=https%3A%2F%2Ffinancial-spider.jurassic.ninja&site_icon&site_lang=en_US&site_created=2022-09-23+08%3A07%3A48&allow_site_connection=1&locale=en&_ui=woo%3AlxDZ4Sce8nm0xpiTaNFN3I6p&_ut=anon&calypso_env=production&_as=wp-cli&from=woocommerce-onboarding&purchase_nonce=qV47s0a6&_wp_nonce=7fe5053a58&redirect_after_auth=https%3A%2F%2Ffinancial-spider.jurassic.ninja%2Fwp-admin%2Fadmin.php%3Fpage%3Dwc-admin&site=https%3A%2F%2Ffinancial-spider.jurassic.ninja)

## Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to 74-gh-woocommerce/team-ghidorah

### CODE REVIEW

**Review by ilyasfoo — CHANGES_REQUESTED** [review]
This is looking awesome, @chihsuan! Nice work. So far I've only reviewed the UI/UX, I haven't really looked into codes yet. The code review will be done after.

> 7. Repeat 2-3 steps
> 8. Observe that the screen matches the design [4ixWMlzrxllx93tSFsCW6k-fi-2656%3A36505](https://href.li/?https://www.figma.com/file/4ixWMlzrxllx93tSFsCW6k/?node-id=2656%3A36505)

I wasn't redirected to the screen in design. Instead, I was redirected to the WPCOM approval screen, and then straight to WooCommerce.com dashboard. Instead, I had to copy and paste the login screen while already being logged in to get to the page.

> Note that if the email address is already used, we should redirect user to the login form. I'll implement it in another PR.

~~I think this may work against us since it could confuse users. Currently, I'm seeing an error `An account with this email already exists. Log in or reset your password.` with appropriate links to solve the problem. If we automatically redirect, they may not know the reason for it and thought that it's a bug for having to type their email twice.~~

Edit: Sorry, I wasn't aware of the comment in Figma and the other PR. Ignore this comment.

### Design feedback

1. Error text <p>image</p>

1. Currently, I only have 2FA using authenticator. Some things I noticed: <p>image</p> P/S: Regarding "hiding it", I meant the "OR"  text, not the button


1. In sign up page, using `Surface duo` device, there's a slight padding/margin issue with "First, let's create your account". <p>image</p>

1. It seems woo logo is not linked to correctly, in console I have a 404 with `https://wpcalypso.wordpress.com/calypso/images/woocommerce-logo-e93433f83860b842c846.svg` <p>image</p>

1. The 👁️ icon is slightly off vertically. I think a slight nudge towards the top should do it <p>image</p>



### Unrelated, but worth pointing:

1. In sign in page, It seems "Forgot password" is pointing to `https://wordpress.com/wp-login.php?action=lostpassword` instead of `https://wpcalypso.wordpress.com/wp-login.php?action=lostpassword`. Should we use a relative address to the current domain? Or maybe we could tackle this when we're building forgot password page.
1. If there's an external app that modifies forms (such as 1Password), it could block the 👁️  icon. Probably not worth to consider fixing since it's a super low impact bug. <br>&nbsp; image

**Comment by chihsuan** [comment]
> I wasn't redirected to the screen in design. Instead, I was redirected to the WPCOM approval screen, and then straight to WooCommerce.com dashboard. Instead, I had to copy and paste the login screen while already being logged in to get to the page.

Ah, @ilyasfoo you're right. If you already logged in, it won't show that page...

I think WCCOM controls that logic so copy-and-paste may be the only way to test that screen for now.

**Comment by chihsuan** [comment]
@ilyasfoo Thanks for the review! ❤️ 

> Design feedback
> 1. Error text
> 2. Currently, I only have 2FA using authenticator. Some things I noticed:
> 3. In sign up page, using Surface duo device, there's a slight padding/margin issue with "First, let's create your account".
> 5. The 👁️ icon is slightly off vertically. I think a slight nudge towards the top should do it

I've fixed these in 3f1438a902f7144f34f0e45fdb50eda98cd36756 and d5f2b9eecc263d76cca7b2c6a1d0feb1d4a500f7. 🙏 

> 4. It seems woo logo is not linked to correctly, in console I have a 404 with https://wpcalypso.wordpress.com/calypso/images/woocommerce-logo-e93433f83860b842c846.svg

This is odd. 🤔 It should be like `https://wpcalypso.wordpress.com/calypso/evergreen/images/woocommerce-logo-e93433f83860b842c846.svg`.

Screen Shot 2022-09-28 at 15 45 27

> In sign in page, It seems "Forgot password" is pointing to https://wordpress.com/wp-login.php?action=lostpassword instead of https://wpcalypso.wordpress.com/wp-login.php?action=lostpassword. Should we use a relative address to the current domain? Or maybe we could tackle this when we're building forgot password page.

I think we should address it when we're building forgot password page because that page is not in calypso now. It will lead to 404 if we use a relative address.

> If there's an external app that modifies forms (such as 1Password), it could block the 👁️ icon. Probably not worth to consider fixing since it's a super low impact bug.

Good catch! I'll see if I can fix it. Otherwise, I'll create a follow-up PR for it.

**Comment by chihsuan** [comment]
@ilyasfoo FIY I've changed the base branch to `ghidorah/nux`.

File: client/blocks/login/two-factor-authentication/security-key-form.jsx [hunk 1]
```diff
@@ -51,9 +56,13 @@ class SecurityKeyForm extends Component {
 								} ) }
 							</p>
 							<p>
-								{ translate(
-									'Insert your security key into your USB port. Then tap the button or gold disc.'
-								) }
+								{ isWoo
+									? translate(
+											'Insert your security key into your USB port, then tap the button or gold disc.'
```
Comment (ilyasfoo) [inline client/blocks/login/two-factor-authentication/security-key-form.jsx:61]: @verofasulo It seems Woo and non-Woo text is slightly different:

`Insert your security key into your USB port, then tap the button or gold disc.` <-- Woo
`Insert your security key into your USB port. Then tap the button or gold disc.` <-- Non-woo

Would it be better if we standardise this?

**Review by ilyasfoo — APPROVED** [review]
@chihsuan Thanks for working on this, @chihsuan! Overall, code looks good, although I'd admit the existing logic is a bit more convoluted than I'd be comfortable with. There are some parts where I feel we'd need to have update the tests since the logic looks brittle:

- client/blocks/login/index.jsx
- client/blocks/login/login-form.jsx

Also, I think adding Woo-specific tests for these may be helpful to prevent future potential breaks:
- client/signup/main.jsx
- client/signup/steps/user/index.jsx

Still, I'd like to hear what you think and I'd agree if you'd like to address them in a follow-up. Pre-approving! 🚢

File: client/blocks/login/two-factor-authentication/security-key-form.jsx (hunk 1 above)
Comment (chihsuan) [reply client/blocks/login/two-factor-authentication/security-key-form.jsx:61]: @ilyasfoo It's Larua's call. I asked a similar question here - OQjzEJlmdgfMzBZxwnCbUn-fi ?node-id=0%3A1

**Comment by chihsuan** [comment]
> Thanks for working on this, @chihsuan! Overall, code looks good, although I'd admit the existing logic is a bit more convoluted than I'd be comfortable with. There are some parts where I feel we'd need to have update the tests since the logic looks brittle:

> Still, I'd like to hear what you think and I'd agree if you'd like to address them in a follow-up.

I absolutely agree we should add more tests! I noticed that our unit test coverage is very low, as you can see below:

- [`client/login/test/`](https://github.com/Automattic/wp-calypso/tree/trunk/client/login/test)
- [`client/blocks/login/test/login-form`](https://github.com/Automattic/wp-calypso/blob/trunk/client/blocks/login/test/login-form.jsx)
- [`client/signup/test`](https://github.com/Automattic/wp-calypso/tree/trunk/client/signup/test)

So I can imagine we need a lot of effort to add unit tests for them. 😥  With our tight time window currently, I'd like to add them later. Thanks. 🙏

**Comment by ilyasfoo** [comment]
@chihsuan I agree 💯 . Let's merge this 

What do you think we if create a follow-up issue with all the details on what tests to create? We'll need to test old use cases plus Woo I'd imagine.

cc @adrianduffell

File: client/blocks/login/two-factor-authentication/security-key-form.jsx (hunk 1 above)
Comment (ilyasfoo) [reply client/blocks/login/two-factor-authentication/security-key-form.jsx:61]: Gotcha, @chihsuan! Thanks for clarifying. Closing this comment

**Pull request merged**